src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -197,12 +198,35 @@ def __enter__(self):
         # Starting a new process in non-fork mode requires to allocate memory. Determine the maximum number of processes
         # based on available memory to avoid memory bursting.
         if not self._use_fork:
-            available_max_worker_count = get_available_max_worker_count()
-            self._n_process = min(self._worker_count, self._nlines, available_max_worker_count)
-            bulk_logger.info(f"Not using fork, process count: {self._n_process}")
+            estimated_available_worker_count = get_available_max_worker_count()
```

Comment:
(Not related to this line)
The table in the PR description is hard to read as the width is limited. Please consider making it better.

Reply:
fixed